### ISSUE TITLE
Order of operations

### ISSUE THREAD

**Issue by tonysparks** [schibsted/jslt#9]
Hello,

I'm not sure if this is as designed or a bug.

```
let x = 1
["h" + $x + 1]
```

I would expect the output to be `h11` but instead the output is `h2`.  Rational, is that `h + $x` be evaluated first, which creates a string `h1` and `h1 + 1` would be a new string: `h11`.

**Comment by larsga** [comment]
Yeah, ideally this should be `h11`. I'm aware of this issue, but it's non-trivial to fix.

I'm also not sure whether it's really good practice of the developer to do this and rely on a particular precedence.

**Comment by larsga** [comment]
[This](http://www.engr.mun.ca/~theo/Misc/exp_parsing.htm#classic) seems to be the solution.
